src/simulator/cylinder.py

```diff
@@ -10,29 +11,44 @@ def __init__(self, radius: float, height: float):
         self.radius = radius
         self.height = height
 
-    def bounding_box(self) -> Box:
+    def bounding_box(self, position: Position) -> Box:
         """Returns smallest enclosing upright Box
-        """
 
+           Calculation the smallest bounding box,function moves
+        the shape to the origin position,then rotates it with the
+        corner points. After that,it calculates the dimensions of
+        the box that is tangent to the farthest corners of the rotation.
+        then returns the shape in last position, and bounding box have position
+        Exactly the same position as shape.
+        """
         # TODO: The following calculation is only for an upright cube
-        # dimension calculations
-        length = self.shape.length
-        height = self.shape.height
-        width = self.shape.width
-
-        # position calculations
-        x = self.position.x
-        y = self.position.y
-        z = self.position.z
-        phi = self.position.phi
-        theta = self.position.theta
-
-        # assign return Arguments
-        bb_cube = Cube(length, height, width)
-        bb_position = Position(x, y, z, phi, theta)
-
-        # return Box(1, bb_cube, bb_position)
-        pass
+        r = self.radius
+        h = self.height
+        # Notice: Because the shape is symmetrical,
+        # only the corner points of top disk are needed, then double the dimensions.
+        points = [[r, 0, h / 2], [0, r, h / 2], [-r, 0, h / 2], [0, -r, h / 2]]
+        new_x = list()
+        new_y = list()
+        new_z = list()
+        for point in points:
+            # Notice: At these angles, theta does not change the shape,
```

Comment:
put the comment inside the if. Here it is not clear what angles you're talking about.